middleware.js

```diff
@@ -149,6 +164,20 @@ module.exports = function(options) {
                 }
               });
             });
+
+            if (sourceMap) {
+              mkdirp(dirname(cssMapPath), '0700', function(err) {
+                if (err) {
+                  return error(err);
+                }
+
+                fs.writeFile(cssMapPath, result.map, 'utf8', function(err) {
```

Comment:
`fs.writeFile(this.options.sourceMap, result.map, 'utf8', function(err))`
after that, you would not need `cssMapPath`.